server/server.go

```diff
@@ -876,6 +878,34 @@ func (a *ArgoCDServer) newHTTPServer(ctx context.Context, port int, grpcWebHandl
 	return &httpS
 }
 
+var extensionsPattern = regexp.MustCompile(`^extension(.*).js$`)
+
+func (a *ArgoCDServer) serveExtensions(extensionsSharedPath string, w http.ResponseWriter) {
+	w.Header().Set("Content-Type", "application/javascript")
+
+	content := ""
+	if err := filepath.Walk(extensionsSharedPath, func(filePath string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() && extensionsPattern.MatchString(info.Name()) {
+			data, err := ioutil.ReadFile(filePath)
+			if err != nil {
+				return err
+			}
+			content = content + fmt.Sprintf(`
+// source: %s/%s 
+`, filePath, info.Name()) + string(data)
+		}
+		return nil
+	}); err != nil && !os.IsNotExist(err) {
```

Comment:
Can we go with long for `if` instead of semicolon? I always get lost on the semicolon-style lines. :-P

Reply:
done